.gitignore

```
@@ -1,5 +1,25 @@
-*~
-
-*.so
-
-extra_deletion
\ No newline at end of file
+# ignore everything
+*
+# except those directories and files:
+!*.go
+!*.h
+!*.yaml
```

```suggestion
!/.golangci.yaml
```
This is the only `.yaml` file in the repo except those in `vendor` dir that are already included.
So, it is better to explicitly track only this file instead of all `.yaml` files.